.circleci/config.yml

```diff
@@ -0,0 +1,146 @@
+      - run:
+          name: generate cache checksum
+          command: find . -name pom.xml | sort | xargs -n 1 cat > /tmp/maven_cached_poms
+      - restore_cache:
+          keys:
+            - maven-repo-v1-{{ .Branch }}-{{ checksum "/tmp/maven_cached_poms" }}
+            - maven-repo-v1-{{ .Branch }}-
+            - maven-repo-v1-
+      - run: mvn -q -s .circleci/m2/settings.xml test-compile
+      - save_cache:
+          paths:
+            - ~/.m2/repository
+          key: maven-repo-v1-{{ .Branch }}-{{ checksum "/tmp/maven_cached_poms" }}
+
+jobs:
+  build:
+    executor: java
+    steps:
+      - checkout
+      - cacheit
+
+  deploy:
+    executor: java
+    steps:
+      - checkout
+      - cacheit
+      - run: |
+          git config --global url."https://${GITHUB_TOKEN}:@github.com/".insteadOf git@github.com:
+      - run: git clone git@github.com:librato/ci-scripts.git ci-scripts
+      - run:
+          name: Deploy snapshot
+          command: |
+            VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+            BRANCH=$(echo -n "${CIRCLE_BRANCH}" | sed -e 's/[^0-9a-zA-Z\._\-]/./g' | tr '[:upper:]' '[:lower:]')
+            NEW_VERSION=${VERSION%%-SNAPSHOT}-$BRANCH-SNAPSHOT
+            mvn versions:set -DnewVersion=$NEW_VERSION
+            mvn -s .circleci/m2/settings.xml package -DskipTests
+            mvn -s .circleci/m2/settings.xml deploy:deploy-file -DgroupId=org.apache.zookeeper \
+              -DartifactId=zookeeper-jute \
+              -Dversion=$NEW_VERSION \
+              -Dpackaging=jar \
+              -Dfile=zookeeper-jute/target/zookeeper-jute-$NEW_VERSION.jar \
+              -DrepositoryId=github \
+              -Durl=https://maven.pkg.github.com/solarwindscloud/maven-releases
+            mvn -s .circleci/m2/settings.xml deploy:deploy-file -DgroupId=org.apache.zookeeper \
+              -DartifactId=zookeeper \
+              -Dversion=$NEW_VERSION \
+              -Dpackaging=jar \
+              -Dfile=zookeeper-server/target/zookeeper-server-$NEW_VERSION.jar \
+              -DrepositoryId=github \
+              -Durl=https://maven.pkg.github.com/solarwindscloud/maven-releases
+
+            .circleci/notify.sh "A new snapshot is ready \`$NEW_VERSION\`"
+
+  release:
+    executor: java
+
+    steps:
+      - checkout
+      - cacheit
+      - run: |
+          git config --global url."https://${GITHUB_TOKEN}:@github.com/".insteadOf git@github.com:
+      - run: git clone git@github.com:librato/ci-scripts.git ci-scripts
+      - run:
+          name: Check versions
+          command: mvn versions:use-releases
+      - run:
+          name: Configure git for release
+          command: |
+            git config user.name "librato-ci"
+            git config user.email "tools+librato-ci-githublibrato.com"
+      - run:
+          name: Perform release
+          command: |
+            VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
```

Comment:
Note, we aren't incrementing the version here, so its assumed that each patch (probably won't be another) will need to manually set their own release version.